07-routing-gossip.md

```diff
@@ -80,13 +80,13 @@ announcement message; that is done by having a signature from each
     * [`64`:`node_signature_2`]
     * [`64`:`bitcoin_signature_1`]
     * [`64`:`bitcoin_signature_2`]
+    * [`2`:`len`]
```

Comment:
As pointed out in my top-level review comment, this assumes that channels are still using the regular 2-of-2 construction. In my mind, this change should be a first step that allows us to be more flexible w.r.t the way channel validation (on both the signature and output checking level) is performed. 

For example, it's possible to (today w/o any further changes to Bitcoin) create funding output that's actually just a regular `p2wkh` output. Both sides are able to in a distributed fashion: conjointly generate multiplicative shares of a private key, and then conduct a 2-party protocol to sign arbitrary messages under the full public key. 

In order to allow things like the above (and other much more exotic constructions), then feature bits should instead be moved to lie either right _below_ the node signatures, or at the very _top_ of the channel announcement message. Moving it to the very top will also let us signal changes to the signature algorithm used in the future. 

Two such possible signature upgrades include: schnorr signatures (to let us aggregate sigs of the same message and save cycles using batch validation _across_ diff messages) and even further pairing based signatures (which would let us aggregate signatures across _batches_ of messages. Both of these signatures schemes could be adopted immediately, the only lagging factor being the maturity of implementations available to the various lightning daemon implementations.